proxy/src/lib.rs

```diff
@@ -229,7 +222,8 @@ where
             let fut = serve(
                 outbound_listener,
                 outgoing,
-                tcp_connect_timeout,
+                config.public_connect_timeout
+                    .unwrap_or_else(||Duration::from_millis(300)),
```

Comment:
The defaulting for things should happen in Config`, not in the users of `Config`.

Reply:
Changed in 675c05b789904c9f90d1fe6598590eb0981a9063